packages/zosfiles/src/methods/copy/Copy.ts

```diff
@@ -255,7 +255,7 @@ export class Copy {
      * @see https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.izua700/IZUHPINFO_API_PutDataSetMemberUtilities.htm
      */
     public static async copyPDS (
-        session: AbstractSession, sourceMemberList: string[], fromPds: string, toPds: string, task: ITaskWithStatus): Promise<IZosFilesResponse> {
+        session: AbstractSession, fromPds: string, toPds: string, task: ITaskWithStatus, sourceMemberList?: string[]): Promise<IZosFilesResponse> {
         try {
 
             if(sourceMemberList.length == 0) {
```

Comment:
We'll need to update this check to make sure that `sourceMemberList` is defined before accessing its `length` property:

```suggestion
            if(sourceMemberList?.length === 0) {
```

Otherwise, the function will throw a TypeError `Failed to read properties of 'undefined' (reading sourceMemberList)`